modules/react/text-input/lib/TextInput.tsx

```diff
@@ -21,74 +14,103 @@ export interface TextInputProps extends Themeable, GrowthBehavior {
   width?: number | string;
 }
 
-const StyledTextInput = styled('input')<
-  Pick<TextInputProps, 'error' | 'grow' | 'width' | 'theme'> & StyledType
->(
-  {
-    ...type.levels.subtext.large,
-    border: `1px solid ${inputColors.border}`,
+export const textInputStencil = createStencil({
+  vars: {
+    width: '',
+  },
+  base: ({width}) => ({
+    ...system.type.subtext.large,
     display: 'block',
-    backgroundColor: inputColors.background,
-    borderRadius: borderRadius.m,
+    border: `${px2rem(1)} solid ${cssVar(system.color.border.input.default)}`,
+    backgroundColor: system.color.bg.default,
+    borderRadius: system.shape.x1,
     boxSizing: 'border-box',
-    height: 40,
+    height: system.space.x10,
     transition: '0.2s box-shadow, 0.2s border-color',
-    padding: space.xxs, // Compensate for border
-    margin: 0, // Fix Safari
+    padding: system.space.x2, // Compensate for border
+    margin: px2rem(0), // Fix Safari
+    width: cssVar(width),
+    minWidth: cssVar(width, calc.add(calc.multiply(system.space.x20, 3), system.space.x10)),
+    color: system.color.text.default,
+    '::-ms-clear': {
+      display: 'none',
+    },
     '&::placeholder': {
```

Comment:
Should think about adding a `placeholder` class as we do with `hover`, `focus`, etc?